pandas/core/indexes/datetimelike.py

```diff
@@ -687,7 +687,7 @@ def intersection(self, other, sort=False):
         self._validate_sort_keyword(sort)
         self._assert_can_do_setop(other)
 
-        if self.equals(other):
+        if self.equals(other) and not self.has_duplicates:
             return self._get_reconciled_name_object(other)
```

Comment:
should we handle the self.equals(other) but non-unique case here?  (like MultIIndex i think)

Reply:
Should be consistent. best would probably be to handle this here.

Reply:
Handled it here now, and called _convert_can_do_setop. Have seen you moved the remaining part to _intersection in #38190, so I will keep this here as it is now that we are consistent